### PR TITLE
feat!: cleanup and complete removal of ether.js

### PR DESCRIPTION
**Pull request by chybisov**
image

image

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,18 +29,3 @@ or
...
-## How to make a release
-
-1. PR with changes
-2. Merge PR into main
-3. Checkout main
-4. `git pull`
-5. `yarn release`
-6. Make sure everything looks good (e.g. in CHANGELOG.md)
-7. `git push --follow-tags`
-8. Done
```
Comment (mathiasmoeller) [inline README.md:44]: Why is this removed?

File: npm.README.md [hunk 2]
```diff
@@ -1,9 +0,0 @@
-### Installation
-
-```
-npm install --save @lifi/types
-```
-
-### Summary
```
Comment (mathiasmoeller) [inline npm.README.md:7]: Why is this file removed?

File: src/api.ts [hunk 3]
```diff
@@ -97,13 +117,22 @@ export interface Route {
   gasCostUSD?: string // Aggregation of underlying gas costs in usd
 
   containsSwitchChain?: boolean // Features required for route execution
-  infiniteApproval?: boolean // Features used for route execution
 
-  steps: LifiStep[]
+  steps: LiFiStep[]
```
Comment (mathiasmoeller) [inline src/api.ts:121]: Even if small, this is a breaking change so make sure to release a major release with breaking changes in the release notes :)

File: src/chains/EVMChain.ts [hunk 4]
```diff
@@ -1,5 +1,4 @@
-import { BigNumber } from 'ethers'
-import { _Chain } from './Chain'
+import type { _Chain } from './Chain.js'
```
Comment (mathiasmoeller) [inline src/chains/EVMChain.ts:1]: why from Chain.js?

File: tsconfig.base.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{
+  // This tsconfig file contains the shared config for the build (tsconfig.build.json) and type checking (tsconfig.json) config.
```
Comment (mathiasmoeller) [inline tsconfig.base.json:2]: comments are not allowed in JSON files, are they?

File: .github/workflows/publish.yaml [hunk 6]
```diff
@@ -6,6 +6,7 @@ on:
       - 'v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+'
       - 'v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+'
       - 'v[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (chybisov) [reply .github/workflows/publish.yaml:9]: I am trying to remember why this was added (didn't work without it?). 😅 We have this in some other repos as well. @dnnnvx please advise 🙂

**Review by mathiasmoeller — COMMENTED** [review]
When this is ready, before we release it, we should also try to integrate it in the BE and see how that goes :)

File: .github/workflows/publish.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (mathiasmoeller) [reply .github/workflows/publish.yaml:9]: This is there to manually trigger a workflow

[the remark this answers is not in the document]
Comment (chybisov) [reply .github/workflows/publish.yaml:9]: @mathiasmoeller Sometimes, triggering workflow manually is useful, especially for alpha/beta releases.

File: npm.README.md (hunk 2 above)
Comment (chybisov) [reply npm.README.md:7]: To simplify the release config, we don't have different READMEs for other repos and maintaining it is just an overhead.

File: src/api.ts (hunk 3 above)
Comment (chybisov) [reply src/api.ts:121]: Yes, there is a bunch of breaking changes and this is one of them 😄

File: src/chains/EVMChain.ts (hunk 4 above)
Comment (chybisov) [reply src/chains/EVMChain.ts:1]: Oh... that's a hot topic and will require a long answer 😂 
TL;DR This is needed for the new package/ts ESM configuration and setup.

Something to read https://github.com/microsoft/TypeScript/issues/49083

File: tsconfig.base.json (hunk 5 above)
Comment (chybisov) [reply tsconfig.base.json:2]: I have no issues with them as part of tsconfig. Do you?

File: .github/workflows/publish.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (chybisov) [reply .github/workflows/publish.yaml]: I removed it for now, if we need it in the future we can always add it back 😄

File: README.md (hunk 1 above)
Comment (chybisov) [reply README.md:44]: Should this be a part of internal documentation or the onboarding process? You don't usually see this in the README of the OSS projects 🙂

Comment (mathiasmoeller) [reply README.md:44]: Sure, you can add it to the  [internal Knowledge Base](https://www.notion.so/lifi/Knowledge-Base-75decf2b0e494555a35ad93a595a2086), Maybe add a new page called `Releasing Packages` or so?

Comment (chybisov) [reply README.md:44]: I added [Releasing Packages](https://www.notion.so/lifi/Releasing-Packages-b22e5e23cc2c482caa57ff977c5fa4fa) page 🙂

**Comment by mathiasmoeller** [comment]
This PR introduces so many changes that it's hard to track what we need to do in the backend.
To break it down I started this change here which is minimal and now I'm integrating that in the backend
https://github.com/lifinance/types/compare/tx-request?expand=1

**Review by mathiasmoeller — APPROVED** [review]
